core/src/main/java/com/github/jsonldjava/utils/JsonUtils.java

```diff
@@ -371,6 +363,56 @@ public static Object fromURL(java.net.URL url, CloseableHttpClient httpClient)
         }
     }
 
+    private static InputStream getJsonLdViaHttpUri(final URL url, final CloseableHttpClient httpClient,
+            CloseableHttpResponse response) throws IOException {
+        final HttpUriRequest request = new HttpGet(url.toExternalForm());
+        // We prefer application/ld+json, but fallback to application/json
+        // or whatever is available
+        request.addHeader("Accept", ACCEPT_HEADER);
+        response = httpClient.execute(request);
+
+        final int status = response.getStatusLine().getStatusCode();
+        if (status != 200 && status != 203) {
+            throw new IOException("Can't retrieve " + url + ", status code: " + status);
+        }
+        // follow alternate document location
+        // https://www.w3.org/TR/json-ld11/#alternate-document-location
+        URL alternateLink = alternateLink(url, response);
+        if (alternateLink != null) {
+            return getJsonLdViaHttpUri(alternateLink, httpClient, response);
+        }
+        return response.getEntity().getContent();
+    }
+
+    private static URL alternateLink(URL url, CloseableHttpResponse response)
+            throws MalformedURLException, IOException {
+        if (response.getEntity().getContentLength() > 0
+                && !response.getEntity().getContentType().getValue().equals("application/ld+json")) {
+            for (Header header : response.getAllHeaders()) {
+                if (header.getName().equalsIgnoreCase("link")) {
+                    String alternateLink = "";
+                    boolean relAlternate = false;
+                    boolean jsonld = false;
+                    for (String value : header.getValue().split(";")) {
+                        if (value.trim().startsWith("<")) {
+                            alternateLink = value.replaceAll("<(.*)>", "$1");
```

Comment:
value.trim is called above, but not on this line.

Reply:
In addition, there doesn't seem to be a need to rely on a regular expression here. The last character should be verified to be a closing brace and then this should just remove the first and last characters.

Reply:
Good catch! Treated with 1229f7353bd1f996aa1f0a6e3e6536f16cb4615b.